js/web/lib/onnxjs/backends/webgl/inference-handler.ts

```diff
@@ -134,16 +195,48 @@ export class WebGLInferenceHandler implements InferenceHandler {
     return this.createTextureDataFromTexture(layout, dataType, texture, tensor);
   }
 
-  /**
-   * Create a TextureData object, using the given texture.
-   * This function does not create new texture. Usually used in scenarios using texture sharing. (eg. Reshape)
-   * @param dataType the tensor data type
-   * @param texture the WebGLTexture object to share
-   * @param tensorId the tensor ID of the shared tensor data
-   */
-  createSharedTextureData(
-      layout: TextureLayout, dataType: Tensor.DataType, texture: WebGLTexture, tensorId?: Tensor.Id): TextureData {
-    return this.createTextureDataFromTexture(layout, dataType, texture, undefined, tensorId);
+  reshapeUnpacked(input: Tensor, reshapedDims: readonly number[]): Tensor {
+    const inputTD = this.getOrCreateTextureData(input, TextureType.unpacked);
+    const newTextureLayout: TextureLayout = {
+      channels: inputTD.channels,
+      height: inputTD.height,
+      width: inputTD.width,
+      // handle reshaping into scalar Tensors
+      shape: reshapedDims.length !== 0 ? reshapedDims : [1],
+      strides: ShapeUtil.computeStrides(reshapedDims),
+      unpackedShape: reshapedDims,
+    };
+    const newTextureData = this.createTextureDataFromTexture(newTextureLayout, input.type, inputTD.texture);
+    return newTextureData.tensor;
+  }
+
+  reshapePacked(input: Tensor, reshapedDims: readonly number[]): Tensor {
+    const inputTD = this.getOrCreateTextureData(input, TextureType.packed);
+
+    // check if the reshape is 'cheap'
+    if (isReshapeCheap(input.dims, reshapedDims)) {
```

Comment:
if the reshape is cheap, should it reuse the reshape unpack function?

Reply:
no. the property of texture layout for unpack and pack is different